server/src/main/java/org/opensearch/common/remote/AbstractClusterMetadataWriteableBlobEntity.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.remote;
+
+import org.opensearch.core.compress.Compressor;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.gateway.remote.ClusterMetadataManifest.UploadedMetadata;
+
+/**
+ * An extension of {@link RemoteWriteableEntity} class which caters to the use case of writing to and reading from a blob storage
+ *
+ * @param <T> The class type which can be uploaded to or downloaded from a blob storage.
+ */
+public abstract class AbstractClusterMetadataWriteableBlobEntity<T> extends RemoteWriteableBlobEntity<T> {
```

Comment:
As `NamedXContentRegistry` is the difference between `RemoteWriteableBlobEntity` and `AbstractClusterMetadataWriteableBlobEntity`, we decide whether to use one vs another based on registry. Is this understanding correct?

Reply:
No it also decouples a corresponding manifest file to be uploaded alongside